### PR TITLE
provider/kubernetes: Pod volumes & env vars

### PR DESCRIPTION
**Pull request by lwander**
Implements pod volumes http://kubernetes.io/docs/user-guide/volumes/ and environment variables. Note, this doesn't allow a user to create volumes that outlive a container, but if one exists (created by kubectl), the user can now attach it to their pod.

### CODE REVIEW

**Comment by lwander** [comment]
@duftler

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/description/servergroup/DeployKubernetesAtomicOperationDescription.groovy [hunk 1]
```diff
@@ -63,6 +64,61 @@ class KubernetesContainerDescription {
...
+  String name
+  KubernetesVolumeSourceType type
+  KubernetesHostPath hostPath
+  KubernetesEmptyDir emptyDir
+  KubernetesPersistentVolumeClaim persistentVolumeClaim
+}
+
+@AutoClone
+@Canonical
+class KubernetesHostPath {
```
Comment (duftler) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/description/servergroup/DeployKubernetesAtomicOperationDescription.groovy]: Does this warrant its own wrapper class?

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/ops/servergroup/DeployKubernetesAtomicOperation.groovy [hunk 2]
```diff
@@ -100,6 +101,50 @@ class DeployKubernetesAtomicOperation implements AtomicOperation<DeploymentResul
...
+                .withReadOnly(volumeSource.persistentVolumeClaim.readOnly)
+            volume.persistentVolumeClaim = res.build()
+            break
+
+          default:
+            return null
+        }
+
+        return volume
+      } ).findAll { it }
```
Comment (duftler) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/ops/servergroup/DeployKubernetesAtomicOperation.groovy]: I think you can replace `.collect()` with `.findResults{}` and drop the `.findAll{}` altogether.

**Comment by duftler** [comment]
LGTM after addressing comments.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/description/servergroup/DeployKubernetesAtomicOperationDescription.groovy (hunk 1 above)
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/description/servergroup/DeployKubernetesAtomicOperationDescription.groovy]: In the future there will be more load balancing/routing options in these classes, so I'm trying to mirror the kubernetes API structure as closely as possible (as they won't be able to make backwards-incompatible changes) so I don't have to refactor this when the features are added.

Comment (duftler) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/description/servergroup/DeployKubernetesAtomicOperationDescription.groovy]: sgtm

**Pull request merged**